### PR TITLE
Improve access of `__annotations__`

### PR DESCRIPTION
**Pull request by MichaelSpece**
## Changes
This is a continuation of #216, but with signed, verified commits.

## Tests
One test was added (in #216).  The last run in #216 passed all tests.

- [ ] `make test` run locally
- [ ] `make fmt` applied
- [ ] relevant integration tests applied

### CODE REVIEW

**Comment by MichaelSpece** [comment]
@nfx GitHub was complaining about unsigned commits so I resynced my fork, which closed #216.

**Review by nfx — CHANGES_REQUESTED** [review]
this looks like too big of a change, can't accept it.

please make sure the PR holds diff only for the relevant changes. most likely the newline characters are changed or something like it.

**Comment by MichaelSpece** [comment]
@nfx Ok; the line endings had indeed been modified.  I have now rebased with auto handling of line endings and the diff is showing a relatively small difference.

File: tests/test_core.py [hunk 1]
```diff
@@ -207,3 +207,19 @@ def test_config_accounts_dod_is_accounts_host(config):
...
+        def __init__(self):
+            super().__init__()
+
+    with pytest.raises(ValueError): # As opposed to `KeyError`.
+        DatabricksConfig()
+
+
+if __name__ == "__main__":
+    import conftest
+    test_config_accounts_dod_is_accounts_host(conftest.config.__pytest_wrapped__.obj())
```
Comment (nfx) [inline tests/test_core.py:225]: these seem irrelevant

**Review by nfx — CHANGES_REQUESTED** [review]

**Pull request merged**